### DISCUSSION TITLE
Access to man pages

### DISCUSSION THREAD

**Discussion by pvervoo** [AppImage/AppImageKit#1344] [Q&A]
I ( had to ) use an appimage and as I didn't know about it, I googled a bit.  I found out it is a competitor for snap and flatpak.  What I like about appimage is its size and speed !  But the app I use has a man page ( under /tmp/.mount_xxxxx/usr/share/man/... ) that is not readily available from the OS.  How can I conveniently consult the man page, embedded in an appimage ?

**Comment by Drsheppard01** [comment]
```<your_appimage> --appimage-extract```

**Reply by pvervoo** [reply]
Yes, of course, but this means that every time you want to read the man page, you have to extract the appimage ( unless you keep it somewhere in your filesystem => overhead ! ).
I want to be able to use the 'regular' man interface i.e.  man <appimage_name>.  Is a kind of link from the man directories to the appimage man pages possible ?

**Comment by TheAssassin** [comment]
> competitor

No, it's an alternative solution that is more portable, doesn't require root permissions runs almost everywhere, too.

Regarding your question, since this is a rootless technology, I'm not sure how we could implement this. I could imagine adding a `--appimage-man` command to the runtime, but then again, what would you do if there is more than one? (Of course, we could require a man page name, too).

Is there a man page location in one's home directory which we could extract the man pages to?

**Comment by probonopd** [comment]
Please keep in mind that AppImage is mainly intended for graphical applications. It works with command line tools, too, but that is really just a coincidence.

**Comment by pdfkungfoo** [comment]
Am Mo., 5. Aug. 2024 um 09:11 Uhr schrieb probonopd <
***@***.***>:

> Please keep in mind that AppImage is mainly intended for graphical
> applications. It works with command line tools, too, but that is really
> just a coincidence.
>
It is stoopid to not make use of such a coincidence if one gets blessed by
it.

**Comment by pvervoo** [comment]
Sorry I missed some advantages 🙂.

I find it a very good idea adding a man option to the command.  If there is no man page, you can show a message.  If there is a man page, you can show it.  If there are several man pages, you can show a list of man pages. In the latter case, the user can add the desired man page name to the man option of the appimage command.


As for the reason for a man page for a graphical application:  If the application would have been installed the 'normal' way, it would have had a man page in the regular OS locations.  I know the average user won't look at the man pages of a graphical application, but the seasoned Linux user does 🙂.

A.f.a.i.k. the OS man command doesn't include searching in one's $HOME.
